src/view/dataElements/xdmObject/helpers/platform.js

```diff
@@ -10,11 +10,35 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-const PROD = "https://platform.adobe.io";
+const PLATFORM_HOST_PROD = "https://platform.adobe.io";
+const PLATFORM_HOST_STAGING = "https://platform-stage.adobe.io";
+
+const IMS_HOST_PREFIX_PROD = "ims-na1";
+const IMS_HOST_PREFIX_STAGING = "ims-na1-stg1";
 
 export default {
-  getHost: () => {
-    return PROD;
+  /**
+   * Returns the platform hostname appropriate for the token IMS environment
+   * @return {string}
+   */
+  getHost: ({ imsAccess }) => {
+    const tokenParts = imsAccess.split(".");
+    if (!tokenParts[1]) {
+      return PLATFORM_HOST_PROD;
+    }
+
+    /**
+     * attempts to read `as` field from access token and use as environment reference
+     * NOTE: assumes production
+     */
+    const environment = imsAccess
+      ? JSON.parse(atob(tokenParts[1])).as
```

Comment:
What do you think about adding a try/catch for the syntax errors that could be thrown from JSON.parse?

Reply:
Probably a good idea